src/collectors/mongodb/test/testmongodb.py

```diff
@@ -133,6 +134,27 @@ def test_should_ignore_unneeded_collections(self,
 
         self.assertPublishedMany(publish_mock, metrics)
 
+    @run_only_if_pymongo_is_available
+    @patch('pymongo.MongoClient')
+    @patch.object(Collector, 'publish')
+    def test_should_publish_replset_status_if_enabled(self,
+                                                      publish_mock,
+                                                      connector_mock):
+        data = {'more_keys': long(1), 'key': 2, 'string': 'str'}
+        self._annotate_connection(connector_mock, data)
+
+        self.collector.collect()
+
+        self.connection.admin.command.assert_any_call('replSetGetStatus')
```

Comment:
Possible to test the opposite -- if replica: False then it *doesnt* call replSetGetStatus?

(e.g. Can you modify the collector config during the test?)